pkgs/by-name/ne/nexusmods-app/package.nix

```diff
@@ -1,114 +1,167 @@
-  };
-in
 buildDotnetModule rec {
-  pname = "nexusmods-app";
+  pname = "nexusmods-app" + lib.strings.optionalString enableUnfree "-unfree";
 
-  version = "0.4.1";
+  version = "0.5.3";
 
   src = fetchFromGitHub {
     owner = "Nexus-Mods";
     repo = "NexusMods.App";
     rev = "v${version}";
     fetchSubmodules = true;
-    hash = "sha256-FzQphMhiC1g+6qmk/R1v4rq2ldy35NcaWm0RR1UlwLA=";
+    hash = "sha256-vy7gc/pS29gphkWM/KezZxXDVsD5DV02b/72pPh2Y2c=";
   };
 
-  projectFile = "NexusMods.App.sln";
+  # If the whole solution is published, there seems to be a race condition where
+  # it will sometimes publish the wrong version of a dependent assembly, for
+  # example: Microsoft.Extensions.Hosting.dll 6.0.0 instead of 8.0.0.
+  # https://learn.microsoft.com/en-us/dotnet/core/compatibility/sdk/7.0/solution-level-output-no-longer-valid
+  # TODO: do something about this in buildDotnetModule
+  projectFile = "src/NexusMods.App/NexusMods.App.csproj";
+  testProjectFile = "NexusMods.App.sln";
 
-  nativeBuildInputs = [
-    copyDesktopItems
-  ];
+  nativeBuildInputs = [ copyDesktopItems ];
 
   nugetDeps = ./deps.nix;
 
   dotnet-sdk = dotnetCorePackages.sdk_8_0;
   dotnet-runtime = dotnetCorePackages.runtime_8_0;
 
+  dotnetBuildFlags = [
+    # From https://github.com/Nexus-Mods/NexusMods.App/blob/v0.5.3/src/NexusMods.App/app.pupnet.conf#L38
+    "--property:Version=${version}"
+    "--property:TieredCompilation=true"
+
+    # From https://nexus-mods.github.io/NexusMods.App/developers/Contributing/#for-package-maintainers
+    # TODO: add DefineConstants support to buildDotnetModule
+    (
+      let
+        constants = [
+          # Tell the app it is a distro package; affects wording in update prompts
+          "INSTALLATION_METHOD_PACKAGE_MANAGER"
+
+          # Don't include upstream's 7zz binary; we use the nixpkgs version
+          "NEXUSMODS_APP_USE_SYSTEM_EXTRACTOR"
+        ];
+      in
+      "--property:DefineConstants=${lib.strings.concatStringsSep "%3B" constants}"
+    )
+  ];
+
   preConfigure = ''
-    substituteInPlace Directory.Build.props \
-      --replace '</PropertyGroup>' '<ErrorOnDuplicatePublishOutputFiles>false</ErrorOnDuplicatePublishOutputFiles></PropertyGroup>'
+    # TODO: Should we patch the desktop file here?
   '';
 
   postPatch = ''
-    ln --force --symbolic "${lib.getExe _7zzWithOptionalUnfreeRarSupport}" src/ArchiveManagement/NexusMods.FileExtractor/runtimes/linux-x64/native/7zz
+    # TODO: Should we patch the desktop file here?
   '';
 
   makeWrapperArgs = [
-    "--prefix PATH : ${lib.makeBinPath [desktop-file-utils]}"
-    "--set APPIMAGE $out/bin/${meta.mainProgram}" # Make associating with nxm links work on Linux
+    # Make associating with nxm links work on Linux
+    # TODO: is this still needed when we use INSTALLATION_METHOD_PACKAGE_MANAGER?
+    "--set APPIMAGE ${placeholder "out"}/bin/${meta.mainProgram}"
   ];
 
+  propagatedBuildInputs = [ (_7zz.override { inherit enableUnfree; }) ];
```

Comment:
If this is just needed at runtime, we might as well leave it up to the user to add it to PATH.

Reply:
We need 7zz at runtime, in the tests and (I believe) at buildtime. Is this the correct way to achieve that?

> we might as well leave it up to the user to add it to PATH.

Not sure what you mean here. Isn't the wrapper responsible for making runtime deps available?

Reply:
Looking in the wrapper script, I can't see any propagated inputs or runtimeDeps being added to the PATH:

```sh
PATH=${PATH:+':'$PATH':'}
PATH=${PATH/':''/nix/store/XXX-dotnet-runtime-8.0.7/bin'':'/':'}
PATH='/nix/store/XXX-dotnet-runtime-8.0.7/bin'$PATH
PATH=${PATH#':'}
PATH=${PATH%':'}
export PATH
```